File: design-patterns/go/iterator/pagination/page.go
// page.go

package main

import "fmt"

type Page struct {
	number int
	path   string
}

func NewPage() (page *Page) {
	page = &Page{}
	return
}
func (page *Page) GetNumber() int {
	return page.number
}
func (page *Page) GetPath() string {
	if page.path == "" {
		return fmt.Sprintf("/page/%v", page.number)
	}
	return page.path
}

func (page *Page) SetNumber(number int) {
	page.number = number
}

func (page *Page) SetPath(path string) {
	page.path = path
}

File: design-patterns/ts/iterator/pagination/page.ts
// page.ts

class Page {
    private pageNumber: number = 0;
    private path: string | null = null;

    getNumber(): number {
        return this.pageNumber;
    }

    setNumber(pageNumber: number): void {
        this.pageNumber = pageNumber;
    }

    getPath(): string {
        if (this.path == null) {
            return "/page/" + this.pageNumber;
        }
        return this.path;
    }

    setPath(path: string): void {
        this.path = path;
    }
}

export default Page;